### PR TITLE
feat: refresh boilerplate deps and update instructions for candidates

### PR DESCRIPTION
**Pull request by skattwood**
Bring the take-home up to current stable versions and tighten the surrounding hygiene so candidates can clone, run, and submit cleanly. Added language around AI tools usage to the instructions. 

## Changes

**Dependency upgrades (latest stable):**
- Next `latest` → `^16.2.6` (pinned), React 18 → 19, `@types/react` → 19
- Jest 29 → 30 + `jest-environment-jsdom` 30 + `@types/jest` 30
- `@testing-library/react` 14 → 16 (+ new peer `@testing-library/dom@^10`)
- TypeScript 5.3 → 5.9, Prettier 3.2 → 3.8
- `@react-google-maps/api` 2.19 → 2.20, `@testing-library/jest-dom` 6.1 → 6.9
- Removed `bootstrap` (was used solely as a CSS reset)

**Forced fixes from the upgrades:**
- Moved `pages/index.test.tsx` → `__tests__/index.test.tsx` (Next 16 type-checks every file under `pages/` as a route module)
- Deleted obsolete create-next-app snapshot file
- `tsconfig.json`: `target` → `ES2022` (was deprecated `es5`), removed deprecated `baseUrl`, `paths` now `./`-prefixed
- Added `next.config.js` whitelisting `*.cloudfront.net` so `next/image` works with SimplyRETS trial photos out of the box

**Boilerplate hygiene:**
- Added `.nvmrc` (Node 22) and `engines.node ">=20.9.0"` (Next 16 minimum)
- Replaced bootstrap-reboot with a minimal inline CSS reset in `styles/global.css`
- Removed `package-lock.json` (yarn-only per instructions)
- `.gitignore`: ignore `/.swc/`, broaden `.env*.local` → `.env*`, restructured AI agent ignores to allow project-level files (`.claude/agents/`, `.claude/skills/`, `.claude/commands/`, `.claude/settings.json`) per the new policy

**INSTRUCTIONS.md:**
- Updated Node requirement (18.17.0 → 20.9.0+; recommends Node 22 LTS)
- Added an **AI Tool Usage Policy** section asking candidates to commit any agent/skill/rules files plus an `AI_USAGE.md` writeup
- Updated Figma mockup link to the current Eng Take-Home Test file
- Added a short note explaining the pre-configured `next/image` host
- Removed the bootstrap mention from the bundled-packages list

## Notes for Reviewers

**Verification done locally:**
- `yarn install` clean (only benign bootstrap-popper warning eliminated by removing bootstrap)
- `yarn build` succeeds on Next 16 + React 19 + TS 5.9
- `yarn test:ci` passes (Jest 30 + RTL 16)
- `yarn dev` boots, `/` and `/listings` return 200, no console errors
- Heart SVGs serve from `/public`
- SimplyRETS API auth + endpoint verified (returns CloudFront photo URLs — pattern in `next.config.js` matches)

**Reviewer to confirm:**
- New Figma link is accessible to candidates (not gated to Side employees)
- Google Maps localhost-restricted API key is still valid

## **Follow-ups for team discussion (not in this PR):**

After upgrading the boilerplate, a few observations about the assignment design itself surfaced — flagging here for separate discussion:

1. **Scope vs 3-hour timebox.** Card grid + map + bounds-driven filtering + localStorage caching + favorites + responsive 375→1680 + tests + README is realistically 4–6 hours for mid-level. Consider tiering features into must-have / nice-to-have to give candidates a priority signal.
2. **Map-bounds filtering may be senior-leaning.** Listening to bounds, debouncing, async filter UX is a real time sink. For mid-level evaluation, "render API data into responsive cards + favorite toggle + tests" might reveal core React skill more cleanly. Bounds-filter as a stretch goal.
3. **Tighten test guidance.** "Add any tests to satisfy confidence" leads to trivial render tests. Be explicit (e.g., "2–3 tests covering an interaction, formatting, and an edge case").
4. **Clarify `react-query` positioning.** Currently "is okay" — is it allowed, recommended, expected? Mid-levels often need the signal.
5. **Trim assessment criteria.** Currently grading on 8 axes simultaneously (functional + production code + Figma fidelity + responsive + a11y + zero warnings + tests + React knowledge). Naming the top 3 gives candidates and reviewers shared calibration.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+22
```
Comment (bporcelli) [inline .nvmrc]: Just curious: Why 22 instead of 24?

File: package.json [hunk 2]
```diff
@@ -9,21 +12,21 @@
     "test:ci": "jest --ci"
   },
   "dependencies": {
-    "@react-google-maps/api": "2.19.3",
-    "bootstrap": "5.3.3",
-    "next": "latest",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "@react-google-maps/api": "^2.20.8",
```
Comment (bporcelli) [inline package.json]: Totally cool with deviating since this is just a take home repo, but our standard practice is to pin deps to specific versions.

File: INSTRUCTIONS.md [hunk 3]
```diff
@@ -110,13 +126,12 @@ In this boilerplate, you will find:
...
   following packages:
   - `prettier`: simple code formating
-  - `bootstrap`: styles reset baseline
   - `@react-google-maps/api`: for rendering google maps
 
 ## Design Mockup
 
 For the mockup, use [this Figma
-design](https://www.figma.com/file/ueqyfCFkBzg14SAoBUGqaN/Side-Take-Home---Nextjs-SimplyRETS).
+design](https://www.figma.com/design/FXPBY3hFfCfl8ZM8SmrbLV/Eng-Take-Home-Test).
```
Comment (bporcelli) [inline INSTRUCTIONS.md:134]: Looks like access to this file is locked down. Do we need to open it up?

**Review by bporcelli — APPROVED** [review]
LGTM - I was able to set up the project locally and get a map rendering using the provided API key. Only thing to call out is the Figma link doesn't appear to be publicly accessible.

File: INSTRUCTIONS.md (hunk 3 above)
Comment (skattwood) [reply INSTRUCTIONS.md:134]: Good catch, it's going to be password protected. Will add the pw to the instructions.

File: .nvmrc (hunk 1 above)
Comment (skattwood) [reply .nvmrc]: It's what Claude decided to go with when I did a full dependency update:

"I picked Node 22 reflexively as the "safer LTS" — it's been out longer, more battle-tested in the wild, and broader CI/tool ecosystem familiarity." 

 Will bump to 24.

File: package.json (hunk 2 above)
Comment (skattwood) [reply package.json]: OK, I will just pin them. Hopefully we are only in the hiring process for a few weeks. We can manually update them before we hire again.

**Review by bporcelli — APPROVED** [review]
